cloudinit/config/schema.py

```diff
@@ -1457,10 +1472,37 @@ def handle_schema_args(name, args):
             "Invalid flag combination. Cannot use --annotate with --docs",
             sys_exit=True,
         )
-    full_schema = get_schema(schema_type="cloud-config")
-    if args.docs:
-        print(load_doc(args.docs))
-        return
+
+
+def get_config_paths_from_args(
+    args,
+) -> Tuple[str, Tuple[Tuple[str, str], ...]]:
+    def get_processed_or_fallback_path(
+        paths: Paths,
+        primary_path_key: str,
+        raw_fallback_path_key: str,
+    ) -> str:
+        """Get processed data path when non-empty of fallback to raw data path.
+        - When primary path isn't set, return empty string as Paths structure
```

Comment:
You are correct. I've dropped this comment. Prior to cloud-init running, Paths will actually default to an instance-id of `iid-datasource-none` instead of as I supposed with the comment. This leaves our paths.get_ipath("userdata") call with a paths response such as `/var/lib/cloud/instances/iid-datasource-none/user-data.txt.i`. In the case where cloud-init hasn't run yet, (or we just ran cloud-init clean) we'll end up with an error message like:

 `Error: Config file /var/lib/cloud/instances/iid-datasource-none/cloud-config.txt does not exist`

I'll drop this comment as it's incorrect